<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Combine &amp; Normalize — combineNormalize • SCWorkflow</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Combine &amp; Normalize — combineNormalize"><meta property="og:description" content="Scales and Normalizes data, Combines samples,
runs Dimensional Reduction, Clusters, and returns a combined Seurat Object."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SCWorkflow</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Combine &amp; Normalize</h1>
    
    <div class="hidden name"><code>combineNormalize.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Scales and Normalizes data, Combines samples,
runs Dimensional Reduction, Clusters, and returns a combined Seurat Object.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">combineNormalize</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  npcs <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  SCT.level <span class="op">=</span> <span class="st">"Merged"</span>,</span>
<span>  vars.to.regress <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  nfeatures <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>  low.cut <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  high.cut <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  low.cut.disp <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  high.cut.disp <span class="op">=</span> <span class="fl">1e+05</span>,</span>
<span>  selection.method <span class="op">=</span> <span class="st">"vst"</span>,</span>
<span>  only.var.genes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  draw.umap <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  draw.tsne <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  seed.for.pca <span class="op">=</span> <span class="fl">42</span>,</span>
<span>  seed.for.tsne <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  seed.for.umap <span class="op">=</span> <span class="fl">42</span>,</span>
<span>  clust.res.low <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  clust.res.high <span class="op">=</span> <span class="fl">1.2</span>,</span>
<span>  clust.res.bin <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  methods.pca <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Elbow"</span>, <span class="st">"Marchenko-Pastur"</span><span class="op">)</span>,</span>
<span>  var.threshold <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  jackstraw <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  jackstraw.dims <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  exclude.sample <span class="op">=</span> <span class="st">""</span>,</span>
<span>  project.name <span class="op">=</span> <span class="st">"scRNAProject"</span>,</span>
<span>  cell.hashing.data <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>object</dt>
<dd><p>a list of seurat objects for each sample.</p></dd>


<dt>npcs</dt>
<dd><p>Select the number of principal components for your analysis.
Please see the elbow plot in the previous template to figure out what
number of PCs explains your variance cut-off. For example,
if the elbow plot has point at (15,0.02), it means that 15 PCs
encapsulate 98% of the variance in your data.(Default: 30)</p></dd>


<dt>SCT.level</dt>
<dd><p>Select at which stage to apply SCtransform nomalization.
Merged: Merge all samples and apply SCTransfrom on merged object.
Sample: Apply SCTranform on individual samples then merge into single
Seurat object. (Default: "Merged")</p></dd>


<dt>vars.to.regress</dt>
<dd><p>Subtract (‘regress out’) this source of heterogeneity
from the data. For example, to Subtract mitochondrial effects,
input "percent.mt." Options: percent.mt, nCount.RNA,
S.Score, G2M.Score, CC.Difference. (Default: NULL)</p></dd>


<dt>nfeatures</dt>
<dd><p>Number of variable features. (Default: 2000)</p></dd>


<dt>low.cut</dt>
<dd><p>Set low cutoff to calculate feature means in
Seurat::FindVariableFeatures. (Default: 0.1)</p></dd>


<dt>high.cut</dt>
<dd><p>Set high cutoff to calculate feature means in
Seurat::FindVariableFeatures. (Default: 8)</p></dd>


<dt>low.cut.disp</dt>
<dd><p>Set low cutoff to calculate feature dispersions in
Seurat::FindVariableFeatures.(Default: 1)</p></dd>


<dt>high.cut.disp</dt>
<dd><p>Set high cutoff to calculate feature dispersions in
Seurat::FindVariableFeatures. (Default: 100000)</p></dd>


<dt>selection.method</dt>
<dd><p>Method to choose top variable features.
Options: vst, mean.var.plot, dispersion. (Default: 'vst')</p></dd>


<dt>only.var.genes</dt>
<dd><p>If dataset is larger than ~40k filtered cells,
set to TRUE. If TRUE, only variable genes will be available for
downstream analysis. (Default: FALSE)</p></dd>


<dt>draw.umap</dt>
<dd><p>If TRUE, draw UMAP plot. (Default: TRUE)</p></dd>


<dt>draw.tsne</dt>
<dd><p>If TRUE, draw TSNE plot. (Default: TRUE)</p></dd>


<dt>seed.for.pca</dt>
<dd><p>Set a random seed for PCA calculation. (Default: 42)</p></dd>


<dt>seed.for.tsne</dt>
<dd><p>Set a random seed for TSNE calculation. (Default: 1)</p></dd>


<dt>seed.for.umap</dt>
<dd><p>Set a random seed for UMAP calculation. (Default: 42)</p></dd>


<dt>clust.res.low</dt>
<dd><p>Select minimum resolution for clustering plots.
The lower you set this, the FEWER clusters will be generated.
(Default: 0.2)</p></dd>


<dt>clust.res.high</dt>
<dd><p>Select the maximum resolution for clustering.
The higher you set this number, the MORE clusters you will produced.
(Default: 1.2)</p></dd>


<dt>clust.res.bin</dt>
<dd><p>Select the bins for your cluster plots.
For example, if you input 0.2 as your bin, and have low/high resolution
ranges of 0.2 and 0.6, then the template will produce cluster plots
at resolutions of 0.2, 0.4 and 0.6. (Default: 0.2)</p></dd>


<dt>methods.pca</dt>
<dd><p>Methods available: Marchenko-Pastur: use eigenvalue
null upper bound from URD, Elbow: Find threshold where percent change
in variation between consecutive PCs is less than X% (set in var.threshold).
(Default: c('Elbow','Marchenko-Pastur'))</p></dd>


<dt>var.threshold</dt>
<dd><p>For Elbow method, set percent change threshold
in variation between consecutive PCs. (Default: 0.1)</p></dd>


<dt>jackstraw</dt>
<dd><p>Opt to visualize your data in a Jackstraw plot. Jackstraw
plot can add more description than an elbow plot but is compute intensive
process and may not be suitable for larger datasets. (Default: FALSE)</p></dd>


<dt>jackstraw.dims</dt>
<dd><p>Recommended max 10.(Default: 5)</p></dd>


<dt>exclude.sample</dt>
<dd><p>Exclude unwanted samples from the merge step.
Include sample names to be removed. If you want to exclude several samples,
separate each sample number by comma (e.g. sample1,sample2,sample3,sample4).</p></dd>


<dt>project.name</dt>
<dd><p>Add project name to the Seurat object metadata.
(Default: 'scRNAProject')</p></dd>


<dt>cell.hashing.data</dt>
<dd><p>Set to "TRUE" if you are using cell-hashed data.
(Default: FALSE)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Seurat Objects and QC plots</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This is Step 3 in the basic Single-Cell RNA-seq workflow.
This template will summarize the multi-dimensionality of your data into
a set of "principal components" to allow for easier analysis.</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Maggie Cam, Thomas Meyer, Jing Bian, Alexandra Michalowski, Alexei Lobanov, Philip Homan, Rui He.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

