<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot coexpression of 2 markers using transcript and/or protein
expression values — dualLabeling • SCWorkflow</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot coexpression of 2 markers using transcript and/or protein
expression values — dualLabeling"><meta property="og:description" content="This method provides visualization of coexpression of 2 genes
(or proteins) and additional methods for filtering for cells with gene
expression values that are above or below thresholds set for one or both
markers. The method allows for filtering (optional) of the Seurat object
using manually set expression thresholds."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SCWorkflow</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot coexpression of 2 markers using transcript and/or protein
expression values</h1>
    
    <div class="hidden name"><code>dualLabeling.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This method provides visualization of coexpression of 2 genes
(or proteins) and additional methods for filtering for cells with gene
expression values that are above or below thresholds set for one or both
markers. The method allows for filtering (optional) of the Seurat object
using manually set expression thresholds.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">dualLabeling</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  <span class="va">samples</span>,</span>
<span>  <span class="va">marker.1</span>,</span>
<span>  <span class="va">marker.2</span>,</span>
<span>  marker.1.type <span class="op">=</span> <span class="st">"SCT"</span>,</span>
<span>  marker.2.type <span class="op">=</span> <span class="st">"SCT"</span>,</span>
<span>  data.reduction <span class="op">=</span> <span class="st">"umap"</span>,</span>
<span>  point.size <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  point.shape <span class="op">=</span> <span class="fl">16</span>,</span>
<span>  point.transparency <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  add.marker.thresholds <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  marker.1.threshold <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  marker.2.threshold <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  filter.data <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  M1.filter.direction <span class="op">=</span> <span class="st">"greater than"</span>,</span>
<span>  M2.filter.direction <span class="op">=</span> <span class="st">"greater than"</span>,</span>
<span>  apply.filter.1 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  apply.filter.2 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  filter.condition <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  parameter.name <span class="op">=</span> <span class="st">"Marker"</span>,</span>
<span>  trim.marker.1 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  trim.marker.2 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pre.scale.trim <span class="op">=</span> <span class="fl">0.99</span>,</span>
<span>  density.heatmap <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  display.unscaled.values <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>object</dt>
<dd><p>Seurat-class object</p></dd>


<dt>samples</dt>
<dd><p>Samples to be included in the analysis</p></dd>


<dt>marker.1</dt>
<dd><p>First gene/marker for coexpression analysis</p></dd>


<dt>marker.2</dt>
<dd><p>Second gene/marker for coexpression analysis</p></dd>


<dt>marker.1.type</dt>
<dd><p>Slot to use for first marker. Choices are "SCT",
"protein","HTO" (default is "SCT")</p></dd>


<dt>marker.2.type</dt>
<dd><p>Slot to use for second marker. Choices are "SCT",
"protein","HTO" (default is "SCT")</p></dd>


<dt>data.reduction</dt>
<dd><p>Dimension Reduction method to use for image. Options
are "umap" or "tsne" (default is "umap")</p></dd>


<dt>point.size</dt>
<dd><p>Point size for image (default is 0.5)</p></dd>


<dt>point.shape</dt>
<dd><p>Point shape for image (default is 16)</p></dd>


<dt>point.transparency</dt>
<dd><p>Point transparency for image (default is 0.5)</p></dd>


<dt>add.marker.thresholds</dt>
<dd><p>Add marker thresholds on plot (default is FALSE)</p></dd>


<dt>marker.1.threshold</dt>
<dd><p>Threshold set for first marker (default is 0.5)</p></dd>


<dt>marker.2.threshold</dt>
<dd><p>Threshold set for second marker (default is 0.5)</p></dd>


<dt>filter.data</dt>
<dd><p>Add new parameter column to metadata annotating where
marker thresholds are applied (default is TRUE)</p></dd>


<dt>M1.filter.direction</dt>
<dd><p>Annotate cells that have gene expression levels
for marker 1 using the marker 1 threshold. Choices are "greater than"
or "less than" (default is "greater than")</p></dd>


<dt>M2.filter.direction</dt>
<dd><p>Annotate cells that have gene expression levels
for marker 2 using the marker 2 threshold. Choices are "greater than"
or "less than" (default is "greater than")</p></dd>


<dt>apply.filter.1</dt>
<dd><p>If TRUE, apply the first filter (default is TRUE)</p></dd>


<dt>apply.filter.2</dt>
<dd><p>If TRUE, apply the second filter (default is TRUE)</p></dd>


<dt>filter.condition</dt>
<dd><p>If TRUE, apply both filters 1 and 2 and take
intersection. If FALSE, apply both filters and take the union.</p></dd>


<dt>parameter.name</dt>
<dd><p>Name for metadata column for new marker filters
(Default is "Marker")</p></dd>


<dt>trim.marker.1</dt>
<dd><p>Trim top and bottom percentile of marker 1 signal to
pre-scale trim values (below) to remove extremely low and high values
(Default is TRUE)</p></dd>


<dt>trim.marker.2</dt>
<dd><p>Trim top and bottom percentile of marker 2 signal to
pre-scale trim values (below) to remove extremely low and high values
(Default is TRUE)</p></dd>


<dt>pre.scale.trim</dt>
<dd><p>Set trimming percentile values (Defalut is 0.99)</p></dd>


<dt>density.heatmap</dt>
<dd><p>Creates a additional heatmap showing the density
distribution of cells. (Default is FALSE)</p></dd>


<dt>display.unscaled.values</dt>
<dd><p>Set to TRUE if you want to view the unscaled
gene/protein expression values (Default is FALSE)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a seurat object with optional additional metadata for cells that are
positive or negative for gene markers, a coexpression plot and contingency
table showing sum of cells filtered.</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Maggie Cam, Thomas Meyer, Jing Bian, Alexandra Michalowski, Alexei Lobanov, Philip Homan, Rui He.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

